src/script/user/UserRepository.ts

```diff
@@ -323,6 +333,14 @@ export class UserRepository extends TypedEventEmitter<Events> {
    * Will update the user both in database and in memory.
    */
   private async updateUser(userId: QualifiedId, user: Partial<UserRecord>, isWebSocket = false): Promise<User> {
+    if (user.deleted && user.name) {
+      const dbUser = await this.userService.loadUserFromDb(userId);
+
+      if (dbUser && dbUser.name) {
+        user.name = dbUser?.name;
```

Comment:
This dbUser.name will exist here :)
```suggestion
        user.name = dbUser.name;
```